### PR TITLE
Add validate session API

### PR DESCRIPTION
**Pull request by balloob**
## Proposed change

Add a new API endpoint `POST /api/ingress/validate_session` that will extend the validation of the current session and returns a 401 if it is no longer valid.

I couldn't get auth working in the API test, so decided to stub it out. Not sure how the other API tests do it. Still verify that we call the check auth function.

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (which adds functionality to the supervisor)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue:
- Link to documentation pull request: https://github.com/home-assistant/developers.home-assistant/pull/714

## Checklist

- [ ] The code change is tested and works locally.
- [ ] Local tests pass. **Your PR cannot be merged unless tests pass**
- [ ] There is no commented out code in this PR.
- [ ] I have followed the [development checklist][dev-checklist]
- [ ] The code has been formatted using Black (`black --fast supervisor tests`)
- [ ] Tests have been added to verify that the new code works.

If API endpoints of add-on configuration are added/changed:

- [ ] Documentation added/updated for [developers.home-assistant.io][docs-repository]

[dev-checklist]: https://developers.home-assistant.io/docs/en/development_checklist.html
[docs-repository]: https://github.com/home-assistant/developers.home-assistant

### CODE REVIEW

File: supervisor/api/ingress.py [hunk 1]
```diff
@@ -78,6 +78,19 @@ async def create_session(self, request: web.Request) -> Dict[str, Any]:
...
+    @api_process
+    async def validate_session(self, request: web.Request) -> Dict[str, Any]:
+        """Create a new session."""
+        self._check_ha_access(request)
+
+        # Check Ingress Session
+        session = request.cookies.get(COOKIE_INGRESS)
+        if not self.sys_ingress.validate_session(session):
+            _LOGGER.warning("No valid ingress session %s", session)
+            raise HTTPUnauthorized()
```
Comment (ludeeus) [inline supervisor/api/ingress.py:90]: This needs to be excepted in `api_process` so we can return it like all other API calls

Comment (balloob) [reply supervisor/api/ingress.py:90]: Are you sure? It's already there for the other ingress endpoints.

Also realized that I do it now based on cookies, rewriting it to JSON data like other APIs.

Comment (ludeeus) [reply supervisor/api/ingress.py:90]: For the other calls that raise those, they are not using `api_process`

Comment (balloob) [reply supervisor/api/ingress.py:94]: In `security.py`, the security middleware also just raises this error. It's a security incident, not an error.

Comment (pvizeli) [reply supervisor/api/ingress.py:94]: I don't know how do you want to handle this on the frontend. If you want like /ingress/session then yes, You need a response with an APIError().

File: supervisor/api/ingress.py [hunk 2]
```diff
@@ -78,6 +78,19 @@ async def create_session(self, request: web.Request) -> Dict[str, Any]:
...
+
+        return {}
```
Comment (pvizeli) [inline supervisor/api/ingress.py:96]: is not need. No return generate a `{"status": "ok", "data": null}`

**Review by ludeeus — APPROVED** [review]

**Pull request merged**